src/emulator.c

```diff
@@ -1326,6 +1520,12 @@ void ofono_emulator_set_indicator(struct ofono_emulator *em,
 	gboolean call;
 	gboolean callsetup;
 	gboolean waiting;
+	struct ofono_emulator *em = __ofono_atom_get_data(atom);
+
+	if (!valid_indication(em, atom, name))
```

Comment:
I know. However I don't like us passing the atom in ofono_emulator_set_indicator as the name indicates it operates on a struct ofono_emulator rather than a ofono_atom.